json/json-derivation/src/main/scala/io/sphere/json/generic/package.fmpp.scala

```diff
@@ -46,12 +46,23 @@ package object generic extends Logging {
 
   /** Creates a FromJSON instance for an Enumeration type that encodes the `toString`
     * representations of the enumeration values. */
-  def fromJsonEnum(e: Enumeration): FromJSON[e.Value] = new FromJSON[e.Value] {
-    def read(jval: JValue): ValidatedNel[JSONError, e.Value] = jval match {
-      case JString(s) => e.values.find(_.toString == s).toValidNel(
-        JSONParseError("Invalid enum value: '%s'. Expected one of: %s".format(s, e.values.mkString("'", "','", "'")))
-      )
-      case _ => jsonParseError("JSON String expected.")
+  def fromJsonEnum(e: Enumeration): FromJSON[e.Value] = {
+    // We're using an AnyRefMap for performance, it's not for mutability.
+    val validRepresentations = AnyRefMap(
+      e.values.toList.map { value =>
+        value.toString -> value.validNel[JSONError]
+      }:_*
```

Comment:
To avoid building intermediate `List` instances:
```suggestion
    val validRepresentations = AnyRefMap.from(
      e.values.iterator.map { value =>
        value.toString -> value.validNel[JSONError]
      }
```

Reply:
This is very close to what I had initially, but unfortunately `AnyRefMap.from` is not available on Scala 2.12.17

Reply:
oh, I did not know about it.
Could you use something like `e.values.iterator.map { ... }.toSeq:_*` then?

Reply:
👍 Good idea. Fixed in 38b78d751b4fcc196d48964f4505284afe093992.